dash_labs/dashdown.py

```diff
@@ -0,0 +1,419 @@
+
+
+def _parse_props(component_props, section):
+    """
+    returns the component property dict specified within a code block.
+
+    For example if this was on the first line of the code block:
+       `app-div-props-{"style": {"maxHeight": "800px"}}`
+    it would return the dict:
+        `{"style": {"maxHeight": "800px"}}`
+    """
+    section_split = section.split(component_props, 1)[-1]
+    matched_paren = 0
+    parsed_dict = []
+    for i in section_split:
+        parsed_dict.append(i)
+        if i == "{":
+            matched_paren += 1
+        if i == "}":
+            matched_paren -= 1
+        if matched_paren == 0:
+            return ast.literal_eval("".join(parsed_dict))
+
+
+def _update_props(
+    props_dict, prohibited_props, section, code_block=None, filename=None
+):
+    """
+    update props that are user-supplied  within a code block
+    """
+    props_dict = copy.deepcopy(props_dict)
+    if "side-by-side-true" in section:
+        props_dict["side_by_side"] = True
+    if "side-by-side-false" in section:
+        props_dict["side_by_side"] = False
+    if "code-first-true" in section:
+        props_dict["code_first"] = True
+    if "code-first-false" in section:
+        props_dict["code_first"] = False
+    if "exec-code-true" in section:
+        props_dict["exec_code"] = True
+    if "exec-code-false" in section:
+        props_dict["exec_code"] = False
+
+    if "code-markdown-props" in section:
+        parsed_props = _parse_props("code-markdown-props-", section)
+        _prohibited_props_check(
+            {"code_markdown_props": parsed_props},
+            prohibited_props,
+            code_block,
+            filename,
+        )
+        props_dict["code_markdown_props"] = {
+            **props_dict["code_markdown_props"],
+            **parsed_props,
+        }
+
+    if "clipboard-props" in section:
+        parsed_props = _parse_props("clipboard-props-", section)
+        _prohibited_props_check(
+            {"clipboard_props": parsed_props}, prohibited_props, code_block, filename
+        )
+        props_dict["clipboard_props"] = {
+            **props_dict["clipboard_props"],
+            **parsed_props,
+        }
+
+    if "app-div-props" in section:
+        parsed_props = _parse_props("app-div-props-", section)
+        _prohibited_props_check(
+            {"app_div_props": parsed_props}, prohibited_props, code_block, filename
+        )
+        props_dict["app_div_props"] = {**props_dict["app_div_props"], **parsed_props}
+    return props_dict
+
+
+def _remove_app_instance(code):
+    """
+    remove the app instance from a code block otherwise app.callback doesn't work
+    todo:  This function is a placeholder. It will fail if there are things like unmatched parens in comments within
+           the app instance or spaces between Dash and (  and probably other things too.
+           Better to use AST module and delete node with app instance.
+    """
```

Comment:
would a regex like this work?
![image](https://user-images.githubusercontent.com/1280389/151920923-8a1cdc7b-7191-49b7-861b-d3d6df834e88.png)
`dash\.Dash\([\s|\S]*\)`